src/components/Page/components/Header/Header.tsx

```diff
@@ -56,6 +56,12 @@ export interface HeaderProps extends TitleProps {
   additionalNavigation?: React.ReactNode;
 }
 
+export function isPrimaryAction(
+  x: PrimaryAction | React.ReactNode,
+): x is PrimaryAction {
+  return !React.isValidElement(x) && x !== undefined;
```

Comment:
This feels a little odd to me - this condition is more "this is not a react node" rather than "this is a primary action", which is alright for now but might get confusing if we want to pass more sorts of things into this. Is there a way to check for action's shape rather than not ReactNode's shape?

Reply:
I agree that it feels a little odd, the reason I did it this way is there are no _required_ properties on `primaryAction`